app/gui/src/controller/searcher.rs

```diff
@@ -1158,6 +1163,21 @@ impl Searcher {
     }
 }
 
+
+// === Searcher helpers ===
+
+fn component_list_builder_with_favorites(
+    groups: Rc<Vec<model::execution_context::ComponentGroup>>,
```

Comment:
I don't think we should pass `Rc<Vec<...>>` as argument anywhere. This is an implementation detial that we keep this `Vec` in an Rc. Either create a type for it or at least a type alias. This is just `ImVec`.